pyproject.toml

```diff
@@ -15,6 +15,9 @@ authors = [{ name = "Anthony Scopatz" }, { email = "scopatz@gmail.com" }]
 maintainers = [{ name = "Anthony Scopatz" }, { email = "scopatz@gmail.com" }]
 license = { text = "BSD 2-Clause License" }
 requires-python = ">=3.9"
+dependencies = [
+    "case-insensitive-dictionary; platform_system=='Windows'",
```

Comment:
Obviously adding the first dependency is a concern. Is it acceptable or would you want to vendor something instead? I originally was planning to use [jaraco.collections.FoldedCaseKeyedDict](https://jaracocollections.readthedocs.io/en/latest/#jaraco.collections.FoldedCaseKeyedDict), which also works, but it brings along a slew of transitive dependencies. For that, I could at least attest to the maintainability (especially given the Tidelift enrollment).